### ISSUE TITLE
Callback function for new friend request / accepted request

### ISSUE THREAD

**Issue by jordansinger** [sungwoncho/has_friendship#9]
How might I go about doing something to the likes of "after_create :notify_user" for the friendships model? Something in general in order to notify users of a new request and the requesting user of an accepted request.

**Comment by sungwoncho** [comment]
You could fork the gem and write the custom code in `HasFreindship::Friendship` model.

Adding callbacks sounds like a good idea. Any suggestions on how to implement it? I am taking PR.

**Comment by t2** [comment]
@sungwoncho @jordansinger I am hitting this now. Is there no proper way to notify someone/see a list of people that have requested to be friends?

For instance, is this not possible?

```
@bob.friend_request(@sally)
@sally.friend_requests #=> [@bob]
```

**Comment by sungwoncho** [comment]
@t2 Sorry for the late reply. I am currently using neither this gem nor Ruby in general, so I can't promise that this feature will be implemented soon. But I will try to find some time on the weekend to look it over.

@Pensarfeo any thoughts on this feature?

**Comment by Pensarfeo** [comment]
@sungwoncho I am in your same situation ;) I have not been working on rails for the last year...
@t2 it seems like a very useful feature, either an after create or what about using blocks? It would not be hard to put a yield in these methods. If any good soul wants to help...

**Comment by t2** [comment]
For now I have solved this with a simple query but performance could be a factor. I may look into this.

```
def friend_requests
    HasFriendship::Friendship.where(status: 'requested', friendable_id: id).map(&:friend)
end
```

**Comment by sbadura** [comment]
https://github.com/sungwoncho/has_friendship/pull/41

**Comment by sungwoncho** [comment]
@sbadura has implemented a solution in #41 and it has been released in [v1.1.0](https://github.com/sungwoncho/has_friendship/blob/master/CHANGELOG.md#110-201756).

Closing this issue.
